pkgs/applications/office/paperwork/paperwork-backend.nix

```diff
@@ -30,48 +31,63 @@ buildPythonPackage rec {
 
   sourceRoot = "source/paperwork-backend";
 
-  # Python 2.x is not supported.
-  disabled = !isPy3k && !isPyPy;
-
   patchPhase = ''
     echo 'version = "${version}"' > src/paperwork_backend/_version.py
     chmod a+w -R ..
     patchShebangs ../tools
   '';
 
   propagatedBuildInputs = [
+    distro
+    gtk3
+    libinsane
+    natsort
+    openpaperwork-core
     pyenchant
-    scikit-learn
-    pypillowfight
     pycountry
-    whoosh
-    termcolor
-    python-Levenshtein
-    libinsane
     pygobject3
     pyocr
-    natsort
-    pkgs.poppler_gi
-    pkgs.gtk3
-    distro
-    openpaperwork-core
+    pypillowfight
+    python-Levenshtein
+    poppler_gi
+    scikit-learn
+    termcolor
+    whoosh
   ];
 
-  preCheck = ''
-    export HOME=$(mktemp -d)
-  '';
+  nativeBuildInputs = [
+    gettext
+    shared-mime-info
+    which
+  ];
+
+  checkInputs = [
+    libreoffice
+    openpaperwork-gtk
+    psutil
+    pytestCheckHook
+  ];
 
-  nativeBuildInputs = [ pkgs.gettext pkgs.which pkgs.shared-mime-info ];
   preBuild = ''
     make l10n_compile
   '';
 
-  checkInputs = [ openpaperwork-gtk psutil pkgs.libreoffice ];
+  disabledTests = [
+    # Disable flaky test on build servers
+    # https://gitlab.gnome.org/World/OpenPaperwork/paperwork/-/issues/1035
+    "test_import_duplicated_docx"
+    # TypeError: Skipped expected string as 'msg' parameter, got 'bool' instead
+    "test_scan"
```

Comment:
this tests only fails when pytestCheckHook. Any idea why?

Reply:
unittestCheckHook works instead. and it is more correct because the makefile uses unittest https://gitlab.gnome.org/World/OpenPaperwork/paperwork/-/blob/develop/paperwork-backend/Makefile#L37